src/inc/lib/e164.php

```diff
@@ -0,0 +1,1156 @@
+<?php declare( strict_types=1 );
+
+namespace Sift_For_WooCommerce\WooCommerce_Actions\Lib;
+
+/**
+ * A simple helper library for E.164 parsing and formatting of phone numbers
+ */
+class E164 {
+	/**
+	 * Return an E.164 formatted phone number from a given phone number
+	 *
+	 * @param string $phone_number A phone number which should only contain numbers representing each digit (vanity phone numbers with letters cause issues).
+	 *
+	 * @return string|null An E.164 formatted number or null if the number could not be parsed.
+	 */
+	public static function from_phone_number( string $phone_number ): ?string {
+		$phone_number        = static::pre_process( $phone_number );
+		$parsed_phone_number = static::parse_phone_number( $phone_number );
```

Comment:
I would check for `static::looks_like_e164_format` and return early if not. That would simplify a lot of the process.

Reply:
this function was only intended to see if the phone number is in a specific shape. It was later changed to just seeing if it begins with a `+`.